src/wasm_opt.rs

```diff
@@ -0,0 +1,95 @@
+use anyhow::{anyhow, ensure, Context, Result};
+use lazy_static::lazy_static;
+use std::{
+    fs,
+    path::{Path, PathBuf},
+    process,
+};
+
+fn wasm_opt_url() -> &'static str {
+    lazy_static! {
+        static ref WASM_OPT_URL: String = {
+            format!(
+                "https://github.com/WebAssembly/binaryen/releases/download/version_{version}/binaryen-version_{version}-{arch}-{os}.tar.gz",
+                version = "105",
+                arch = platforms::TARGET_ARCH,
+                os = platforms::TARGET_OS,
+            )
+        };
+    }
+
+    &WASM_OPT_URL
+}
+
+pub fn run(
+    binary_path: impl AsRef<Path>,
+    shrink_level: u32,
+    optimization_level: u32,
+    debug_info: bool,
+) -> Result<()> {
+    let wasm_opt = download_wasm_opt()?;
```

Comment:
:information_source: It's often useful to do the as_ref at the very beginning of the function
```suggestion
) -> Result<()> {
    let input_path = binary_path.as_ref();
    let wasm_opt = download_wasm_opt()?;
```